### PR TITLE
Generics checks for parameter passing

### PR DESCRIPTION
**Pull request by NikitaAware**
Thank you for contributing to NullAway!

Please note that once you click "Create Pull Request" you will be asked to sign our [Uber Contributor License Agreement](https://cla-assistant.io/uber/NullAway) via [CLA assistant](https://cla-assistant.io/).

Before pressing the "Create Pull Request" button, please provide the following:

  - [x] A description about what and why you are contributing, even if it's trivial.

  - [x] The issue number(s) or PR number(s) in the description if you are contributing in response to those.

  - [x] If applicable, unit tests.

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 1]
```diff
@@ -422,4 +442,27 @@ public void checkTypeParameterNullnessForAssignabilityForConditionalExpression(
...
   }
+
+  public void compareNullabilityOfGenericParameters(
+      List<Symbol.VarSymbol> formalParams, List<? extends ExpressionTree> actualParams) {
+    if (!config.isJSpecifyMode()) {
+      return;
+    }
+    // Check if the parameters are generics and compare annotations for the actual and formal
+    // parameters
+    for (int i = 0; i < formalParams.size(); i++) {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:465]: We should add a test case where the invoked method has varargs, like these:

https://github.com/uber/NullAway/blob/master/nullaway/src/test/java/com/uber/nullaway/NullAwayVarargsTests.java#L47

We should write something like:

```java
void foo(NullableTypeParam<@Nullable String>... args) { ... }
```
And then _all_ the parameters passed at a call to `foo()` should have `@Nullable String` as their type parameter.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java [hunk 2]
```diff
@@ -177,6 +177,26 @@ private static void reportMismatchedTypeForTernaryOperator(
...
+  private void reportInvalidParametersNullabilityError(
+      Type formalParameter,
+      Type actualParameter,
+      ExpressionTree expressionTree,
+      VisitorState state,
+      NullAway analysis) {
+    ErrorBuilder errorBuilder = analysis.getErrorBuilder();
+    ErrorMessage errorMessage =
+        new ErrorMessage(
+            ErrorMessage.MessageTypes.TYPE_PARAMETER_CANNOT_BE_NULLABLE,
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java]: we'll want a new type of error here

**Review by msridhar — COMMENTED** [review]
couple of quick comments

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java (hunk 1 above)
Comment (NikitaAware) [reply nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java:465]: Thanks. I will update this.

File: nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java (hunk 2 above)
Comment (NikitaAware) [reply nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java]: something like GNERIC_TYPE_PARAMETER_CANNOT_BE_NULLABLE ?

Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java]: Maybe `PASS_NULLABLE_GENERIC`

File: nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java [hunk 3]
```diff
@@ -632,6 +628,35 @@ public void nestedTernary() {
...
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "static class A<T extends @Nullable Object> { }",
+            "  static A<String> sampleMethod(A<A<String>> a1, A<String> a2) {",
+            "     return a2;",
+            "  }",
+            "  static A<@Nullable String> sampleMethodWithVarArgs(A<A<@Nullable String>> a1, A<@Nullable String> a2, @Nullable Object o1, @Nullable Object o2) {",
```
Comment (msridhar) [inline nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java]: This is not using varargs.  You can read about varargs here:

https://www.baeldung.com/java-varargs

**Review by msridhar — APPROVED** [review]
LGTM!  Let's hold off for the moment on opening a PR upstream

**Pull request closed without merging**